invenio_checks/metadata/expressions.py

```diff
@@ -0,0 +1,338 @@
+            else:
+                success = False
+                message = f"Cannot check if {type(left_value)} does not contain a value"
+        # Implement other operators as needed
+        elif self.operator == "^=":
+            if isinstance(left_value, str):
+                success = left_value.startswith(self.right)
+                message = (
+                    None if success else f"Expected {path} to start with {self.right}"
+                )
+            else:
+                success = False
+                message = f"Cannot check if {type(left_value)} starts with a value"
+        elif self.operator == "!^=":
+            if isinstance(left_value, str):
+                success = not left_value.startswith(self.right)
+                message = (
+                    None
+                    if success
+                    else f"Expected {path} not to start with {self.right}"
+                )
+            else:
+                success = False
+                message = (
+                    f"Cannot check if {type(left_value)} doesn't start with a value"
+                )
+        elif self.operator == "$=":
+            if isinstance(left_value, str):
+                success = left_value.endswith(self.right)
+                message = (
+                    None if success else f"Expected {path} to end with {self.right}"
+                )
+            else:
+                success = False
+                message = f"Cannot check if {type(left_value)} ends with a value"
+        elif self.operator == "!$=":
+            if isinstance(left_value, str):
+                success = not left_value.endswith(self.right)
+                message = (
+                    None if success else f"Expected {path} not to end with {self.right}"
+                )
+            else:
+                success = False
+                message = f"Cannot check if {type(left_value)} doesn't end with a value"
+        else:
+            success = False
+            message = f"Unknown operator: {self.operator}"
+
+        return ExpressionResult(success, path, left_value, message)
+
+
+class LogicalExpression(Expression):
+    """Expression for logical operations (AND, OR)."""
+
+    VALID_OPERATORS = ["and", "or"]
+
+    def __init__(self, operator, expressions):
+        """Initialize the logical expression."""
+        if operator not in self.VALID_OPERATORS:
+            raise ValueError(
+                f"Invalid operator: {operator}. Valid operators are {self.VALID_OPERATORS}"
+            )
+        self.operator = operator
+        self.expressions = expressions
+
+    def evaluate(self, record):
+        """Evaluate the logical expression."""
+        results = [expr.evaluate(record) for expr in self.expressions]
+
+        if self.operator == "and":
+            # For AND, success is True only if all expressions succeed
+            success = all(r.success for r in results)
+            # Return the first failure, if any
```

Comment:
so if there is three failing conditions, the user has to save the draft three times to fix the issue?